lib/practice_exercises.rb

```diff
@@ -1,13 +1,48 @@
-
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(1)
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  return true if string.length < 2
+  
+  i = 0
+  end_i = string.length - 1
+  pattern = /[a-z0-9]/
+  
+  while i < end_i
+    
+    i+= 1 until (string[i] == nil || pattern.match(string[i].downcase))
+    return false if string[i] == nil
+
+    end_i -= 1 until pattern.match(string[end_i].downcase)
+
+    return false if string[i].downcase != string[end_i].downcase
+
+    i += 1
+    end_i -= 1
+  end
+
+  return true
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(1)
 def longest_prefix(strings)
```

Comment:
I would say the time complexity is O(n * m) where m is the length of the shortest word and n being the number of strings.  Otherwise 👍